packages/server/lib/controllers/config.controller.ts

```diff
@@ -55,7 +55,10 @@ class ConfigController {
                     creationDate: config.created_at
                 };
                 if (template && template.auth_mode !== AuthModes.App && template.auth_mode !== AuthModes.Custom) {
-                    integration['connectionConfigParams'] = parseConnectionConfigParamsFromTemplate(template!);
+                    integration['connectionConfigParams'] = parseConnectionConfigParamsFromTemplate(template!).filter(
+                        // we ignore connection config param that are in the token response metadata
+                        (element) => (template.token_response_metadata || []).indexOf(element) == -1
+                    );
```

Comment:
@khaliqgant this is a solution to deal with optional connection config parameter. aka filtering out connection config param if it exists in token response metadata. I am not sure if the `/api/v1/integration` is used somewhere else though.
Another solution would be for this endpoint to return both the connection config params and the token response metadata and let the webapp do the filtering. What do you think?